components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/DefaultServiceURLBuilder.java

```diff
@@ -0,0 +1,212 @@
+            if (StringUtils.isNotBlank(tenantDomain) &&
+                    !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equalsIgnoreCase(tenantDomain)) {
+                resolvedUrlStringBuilder.append("/t/").append(tenantDomain);
+            }
+        }
+
+        if (StringUtils.isBlank(resolvedUrlContext)) {
+            resolvedUrlContext = resolveUrlFromServerConfig(urlContext);
+        }
+
+        if (StringUtils.isNotBlank(resolvedUrlContext)) {
+            resolvedUrlStringBuilder.append(resolvedUrlContext);
+        }
+
+        if (!isTenantQualifiedUrlsEnabled() && appendTenantAsQueryParam) {
+            if (StringUtils.isNotBlank(tenantDomain) &&
+                    !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equalsIgnoreCase(tenantDomain)) {
+                resolvedUrlStringBuilder.append("?").append(MultitenantConstants.TENANT_DOMAIN).append("=")
+                        .append(tenantDomain);
+            }
+        }
+
+        Map<String, Object> parameters = this.parameters;
+        Map<String, Object> fragments = this.fragments;
+
+        return new ServiceURL(protocol, hostName, port, proxyContext, webContext, resolvedUrlStringBuilder.toString(),
+                parameters, fragments);
+
+    }
+
+    private String fetchProtocol() {
+
+        return CarbonUtils.getManagementTransport();
+    }
+
+    private String fetchHostName() throws URLResolverException {
+
+        String hostName = ServerConfiguration.getInstance().getFirstProperty(IdentityCoreConstants.HOST_NAME);
+        try {
+            if (StringUtils.isBlank(hostName)) {
+                hostName = NetworkUtils.getLocalHostname();
+            }
+        } catch (SocketException e) {
+            throw new URLResolverException("Error while trying to resolve the hostname from the system.", e);
+        }
+        return hostName;
+    }
+
+    private Integer fetchPort() {
+
+        String mgtTransport = CarbonUtils.getManagementTransport();
+        AxisConfiguration axisConfiguration = IdentityCoreServiceComponent.getConfigurationContextService().
+                getServerConfigContext().getAxisConfiguration();
+        int port = CarbonUtils.getTransportProxyPort(axisConfiguration, mgtTransport);
+        if (port <= 0) {
+            port = CarbonUtils.getTransportPort(axisConfiguration, mgtTransport);
+        }
+        return port;
+    }
+
+    private String fetchProxyContextPath() {
+
+        return ServerConfiguration.getInstance().getFirstProperty(IdentityCoreConstants.PROXY_CONTEXT_PATH);
+    }
+
+    private String fetchWebContextRoot() {
+
+        return ServerConfiguration.getInstance().getFirstProperty(IdentityCoreConstants.WEB_CONTEXT_ROOT);
+    }
+
+    private String resolveUrlFromStandardContext(String urlContext) {
+        /**
+         * TODO
+         * resolve URL context from the standard context.
+         *
+         */
+        return urlContext;
+    }
```

Comment:
We might not need this. context is the path, which we will be using to construct the full URL